pkg/generators/enum.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/types"
+)
+
+const tagEnumType = "enum"
+
+type enumValue struct {
+	Name    string
+	Value   string
+	Comment string
```

Comment:
An alternative to keeping track of the comments of each enum value would be to keep track of the comment for the entire enum? E.g.:

`go
type enumType struct {
  Values []*enumValue
  Description string
}

type enumValue struct {
  Name string
  Value string
}
`

The advantage being that the description only needs to be constructed once.

Reply:
The comment of entire enum will be used for at most once. I think a getter would be better